pkg/kubelet/lifecycle/handlers_test.go

```diff
@@ -221,3 +225,203 @@ func TestRunHandlerHttpFailure(t *testing.T) {
+				},
+			},
+			attrs: &PodAdmitAttributes{
+				Pod: makePod("12345678", "foo", "new",
+					map[string]string{kubetypes.ConfigSourceAnnotationKey: kubetypes.FileSource},
+					v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "foo"},
+						},
+						HostPID: true,
+					}),
+			},
+			podAdmitResult: PodAdmitResult{Admit: false},
+		},
+		{
+			name: "Host IPC Allowed",
+			capabilities: capabilities.Capabilities{
+				PrivilegedSources: capabilities.PrivilegedSources{
+					HostIPCSources: []string{kubetypes.ApiserverSource, kubetypes.FileSource},
+				},
+			},
+			attrs: &PodAdmitAttributes{
+				Pod: makePod("12345678", "foo", "new",
+					map[string]string{kubetypes.ConfigSourceAnnotationKey: kubetypes.FileSource},
+					v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "foo"},
+						},
+						HostIPC: true,
+					}),
+			},
+			podAdmitResult: PodAdmitResult{Admit: true},
+		},
+		{
+			name: "Host IPC Disallowed",
+			capabilities: capabilities.Capabilities{
+				PrivilegedSources: capabilities.PrivilegedSources{
+					HostIPCSources: []string{},
+				},
+			},
+			attrs: &PodAdmitAttributes{
+				Pod: makePod("12345678", "foo", "new",
+					map[string]string{kubetypes.ConfigSourceAnnotationKey: kubetypes.FileSource},
+					v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "foo"},
+						},
+						HostIPC: true,
+					}),
+			},
+			podAdmitResult: PodAdmitResult{Admit: false},
+		},
+		{
+			name: "Privilege Container Allowed",
+			capabilities: capabilities.Capabilities{
+				AllowPrivileged: true,
+			},
+			attrs: &PodAdmitAttributes{
+				Pod: makePod("12345678", "foo", "new",
+					map[string]string{kubetypes.ConfigSourceAnnotationKey: kubetypes.FileSource},
+					v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "foo", SecurityContext: &v1.SecurityContext{Privileged: &privileged}},
+						},
+					}),
+			},
+			podAdmitResult: PodAdmitResult{Admit: true},
+		},
+		{
+			name: "Privilege Container Disallowed",
+			capabilities: capabilities.Capabilities{
+				AllowPrivileged: false,
+			},
+			attrs: &PodAdmitAttributes{
+				Pod: makePod("12345678", "foo", "new",
+					map[string]string{kubetypes.ConfigSourceAnnotationKey: kubetypes.FileSource},
+					v1.PodSpec{
+						Containers: []v1.Container{
+							{Name: "foo", SecurityContext: &v1.SecurityContext{Privileged: &privileged}},
+						},
+					}),
+			},
+			podAdmitResult: PodAdmitResult{Admit: false},
+		},
+	}
+
+	capabilityAdmitHandler := NewCapabilityAdmitHandler()
+
+	for _, testCase := range testCases {
+		capabilities.SetForTests(testCase.capabilities)
+		result := capabilityAdmitHandler.Admit(testCase.attrs)
+
+		if result.Admit != testCase.podAdmitResult.Admit {
+			t.Fatalf("%v: expected: %v, got: %v", testCase.name, testCase.podAdmitResult.Admit, result.Admit)
+		}
+	}
+
+}
+
+func makePod(uid types.UID, name, namespace string, annotations map[string]string, spec v1.PodSpec) *v1.Pod {
```

Comment:
I see that all test cases use the same uid, name, and namespace. Would it be better to hard-code them inside of the function? It reduces test cases and also leave only the parts that differ among them.